src/vehicle.cpp

```diff
@@ -5102,7 +5103,9 @@ void vehicle::refresh()
         if( vpi.has_flag( "CAMERA" ) ) {
             camera_epower += vpi.epower;
         }
-        if( vpi.has_flag( VPFLAG_FLOATS ) ) {
+        // The floating cache shouldn't include broken parts;
```

Comment:
`refresh` is called only on part addition/removal and similar. `floating` must necessarily contain both broken and unbroken parts.